### PR TITLE
support ScalaPB package/file-level customization

### PR DESCRIPTION
**Pull request by bjaglin**
See https://scalapb.github.io/docs/customizations

On master, code won't compile if package/file-level options are used to change `flat_package` as only ScalaPB honors the request.

### CODE REVIEW

**Comment by raboof** [comment]
OK TO TEST

**Comment by raboof** [comment]
> On master, code won't compile if package/file-level options are used to change `flat_package` as only ScalaPB honors the request.

Indeed the Akka gRPC code generation currently only allows changing `flat_package` as a 'build-wide' wide option. I think we would definitely want to update the Akka gRPC code generation to take into account package/file-level customizations, but this feature is already useful without it.

File: sbt-plugin/src/sbt-test/gen-scala-server/09-scalapb-customizations/test [hunk 1]
```diff
@@ -0,0 +1,5 @@
+> protocGenerate
+$ exists target/scala-2.12/akka-grpc/main/example/myapp/helloworld/grpc/HelloRequest.scala
+$ exists target/scala-2.12/akka-grpc/main/example/myapp/helloworld/grpc/HelloworldProto.scala
+$ exists target/scala-2.12/akka-grpc/main/example/myapp/helloworld/grpc/GreeterService.scala
+> compile
```
Comment (bjaglin) [inline sbt-plugin/src/sbt-test/gen-scala-server/09-scalapb-customizations/test:5]: just force-pushed as I had forgotten this...

**Review by ignasi35 — APPROVED** [review]
Nice that we're getting this. 


(discussion for a separate PR/issue): I wonder if should eventually deprecate and remove the use of `flat_package` out of the box. We would then fallback to scalapb's behavior and transparently use the options available there.

**Comment by bjaglin** [comment]
>  I wonder if should eventually deprecate and remove the use of `flat_package` out of the box. We would then fallback to scalapb's behavior and transparently use the options available there.

I am not sure it will be required. ScalaPB has promising upcoming changes that will make consumption of JAR-packaged Scala stubs compiled with different options than the ones used for the local project seamless.

See https://github.com/scalapb/scalapb-validate/issues/70 & https://github.com/scalapb/ScalaPB/pull/873 for related discussions.

**Comment by ignasi35** [comment]
The motivation of my comment was to eventually remove from akka-gRPC any access to scalaPB settings since those scalaPB settings are already available via options in the `.proto` file.

So users no longer need to specify `akkaGrpcCodeGeneratorSettings := akkaGrpcCodeGeneratorSettings.value.filterNot(_ == "flat_package")` and, instead, use plain scalapb syntax on the `.proto` file.

Again, I'd rather discuss these further improvements on a separate issue/PR since I'm sure there'll be use cases that require more attention.

**Review by raboof — APPROVED** [review]
Great! Would love to take into account the `flat_package` option better in the Akka gRPC codegen side but that's for another PR, this is already a great improvement.

**Pull request merged**